unstructured/chunking/title.py

```diff
@@ -73,27 +73,53 @@ def chunk_by_title(
     multipage_sections
         If True, sections can span multiple pages. Defaults to True.
     combine_text_under_n_chars
-        Combines elements (for example a series of titles) until a section reaches
-        a length of n characters.
+        Combines elements (for example a series of titles) until a section reaches a length of
+        n characters. Defaults to `max_characters` which combines chunks whenever space allows.
+        Specifying 0 for this argument suppresses combining of small chunks. Note this value is
+        "capped" at the `new_after_n_chars` value since a value higher than that would not change
+        this parameter's effect.
     new_after_n_chars
-        Cuts off new sections once they reach a length of n characters (soft max)
+        Cuts off new sections once they reach a length of n characters (soft max). Defaults to
+        `max_characters` when not specified, which effectively disables any soft window.
+        Specifying 0 for this argument causes each element to appear in a chunk by itself (although
+        an element with text longer than `max_characters` will be still be split into two or more
+        chunks).
     max_characters
         Chunks elements text and text_as_html (if present) into chunks of length
         n characters (hard max)
     """
 
-    if not (
-        max_characters > 0
-        and combine_text_under_n_chars >= 0
-        and new_after_n_chars >= 0
-        and combine_text_under_n_chars <= new_after_n_chars
-        and combine_text_under_n_chars <= max_characters
-    ):
+    # -- validation and arg pre-processing ---------------------------
+
+    # -- chunking window must have positive length --
+    if max_characters <= 0:
+        raise ValueError(f"'max_characters' argument must be > 0, got {max_characters}")
+
+    # -- `combine_text_under_n_chars` defaults to `max_characters` when not specified and is
+    # -- capped at max-chars
+    if combine_text_under_n_chars is None or combine_text_under_n_chars > max_characters:
```

Comment:
It makes sense to set `combine_text_under_n_chars` to `max_characters` when not specified, but when it is specified but is greater than `max_characters`, we are just capping it at max_character without a warning that the user input was capped. Perhaps I need a bit more context on this, but does it make sense to cap the user-inputted value without notifying the user of the change?

Reply:
discussion update: the behavior when `combine_text_under_n_chars` is equal to or greater than `max_characters` is the same, so the capping can be done without notification